### ISSUE TITLE
Show a spinner while searching SD card

### ISSUE THREAD

**Issue by Ryuno-Ki** [codexa/firetext#215]
I'm trying out Firetext again and found the UI … unresponsive, while searching my SD card (containing lots of loadable items).

I'd expect a spinner here (since the absolute number of files is not known a priori).
Moreover, if there are too many files, the app crashes with OOM (another issue?).

**Comment by ferndot** [comment]
In regards to the first issue, I would also expect a spinner there. To fix this, we could add the `spinner()` function, defined [here](https://github.com/codexa/firetext/blob/31ae7f44c2731f9210206c6dae25cbffd79487e0/scripts/firetext.js#L1307-L1313), to [this function](https://github.com/codexa/firetext/blob/31ae7f44c2731f9210206c6dae25cbffd79487e0/scripts/firetext.js#L305) and [this function](https://github.com/codexa/firetext/blob/31ae7f44c2731f9210206c6dae25cbffd79487e0/scripts/cloud/cloud.js#L96).

The second issue is much more difficult to fix, but I would imagine that it will be automatically remedied by #174 (we won't try to display all files at once). Can I split it into another issue for tracking purposes? Also, is it possible for you to send us a copy of your SD card to use as a test case?

I am marking this as a first-fix.

**Comment by Ryuno-Ki** [comment]
Go ahead with splitting.
I'm afraid, I cannot share my SD card's content with you. Private stuff (photos of my family and the like).
But you can bulk it with videos and the like (should be included while searching as well?).

I can look after a directory tree, if that would help you. However, I'm going to redact some entries there, too.

**Comment by ferndot** [comment]
@Ryuno-Ki: I didn't think about that, sorry. Could you share the size of your SD card's contents?

**Comment by Ryuno-Ki** [comment]
Don't bother.

Here's what I'm willing to share:

``` bash
 $ du -h -d 2
3.8M    ./LOST.DIR
32K ./.android_secure
21M ./DCIM/100MZLLA
302M    ./DCIM/101MZLLA
3.5G    ./DCIM/MUSIC
1.6G    ./DCIM/VIDEO
21G ./DCIM/FILES
512K    ./DCIM/FFOS-Sticky-App-Imgs
128K    ./DCIM/FFOS-Drawl-App-Imgs
1.9G    ./DCIM/IMAGES
29G ./DCIM
66M ./.gallery/previews
67M ./.gallery
320K    ./.Trash-14908/info
25M ./.Trash-14908/files
26M ./.Trash-14908
128K    ./.Trash-1000/info
32K ./.Trash-1000/files
32K ./.Trash-1000/expunged
224K    ./.Trash-1000
32K ./downloads/bluetooth
64K ./downloads
5.5M    ./screenshots
64K ./.Trash-2111/info
3.8M    ./.Trash-2111/files
3.9M    ./.Trash-2111
64K ./contacts2xml/20130601020207
224K    ./contacts2xml
32K ./.downloaded
29G .
```

Is that sufficient to you? If you are interested only in certain file formats, let me know.

**Comment by ferndot** [comment]
@Ryuno-Ki: thanks! I will test this by using a bunch of Lorem Ipsum files.

**Comment by ferndot** [comment]
I split this issue out into #218.

**Comment by staritza** [comment]
This issue is fixed, no? Trying to duplicate it but the spinner always runs until the "Documents On This Device" section is loaded. We don't need another spinner in the places you've suggested @joshua-s ?

**Comment by staritza** [comment]
Nvm I spoke to soon. Seems the issue starts after edit mode not when the app is first loaded.

**Comment by ferndot** [comment]
Fixed by a0ead51.
